flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -1095,6 +1097,16 @@ SqlTypeNameSpec ExtendedSqlBasicTypeName() :
             typeAlias = token.image;
             precision = Integer.MAX_VALUE;
         }
+    |
+       <TIMESTAMP_LTZ>
+       {
+           typeAlias = token.image;
+       }
+       precision = PrecisionOpt()
+       {
+            typeName = SqlTypeName.TIMESTAMP_WITH_LOCAL_TIME_ZONE;
+            return new SqlTimestampLtzTypeNameSpec(typeAlias, typeName, precision, getPos());
```

Comment:
can't we just use `SqlAlienSystemTypeNameSpec`? We do the same for `STRING`

Reply:
I think we cannot because `SqlTimestampLtzTypeNameSpec`overrides `unparse` which considered precision(eg. `TIMESTAMP_LTZ(3)`) but  `SqlAlienSystemTypeNameSpec` does not

Reply:
but is this really important? For strings we also don't maintain the original SQL but only the expanded version., right?

Reply:
You're right that  `STRING` and `BYTES` didn't keep precision info(Integer.MAX_VALUE by default),  user can not to use `STRING(3)`, `STRING(10)` , but user need to use `TIMESTAMP_LTZ(3), TIMESTAMP_LTZ(6)` in SQL.

Reply:
we checked the unparsed type in `FlinkDDLDataTypeTest`

Reply:
Ok, makes sense to me. +1 then. I will take care of merging this.